src/main/java/org/jenkinsci/plugins/badge/PublicBadgeAction.java

```diff
@@ -127,11 +128,11 @@ public HttpResponse doCoverageIcon(StaplerRequest req, StaplerResponse rsp, @Que
         Job<?, ?> project = getProject(job);
         Integer codeCoverage = null;
 
-        if (project.getLastSuccessfulBuild() != null) {
```

Comment:
Agreed! My view was that the code coverage could have changed since the last successful build, and it may be better to show something like "n/a" instead of a stale value. I updated the code coverage badge to default to "n/a" instead of "0" when a build fails or aborts. If you feel this behavior is not right, please feel free to change back. However, I do believe we should still use the last completed build for all other badges regardless.